packages/@aws-cdk/aws-certificatemanager/lambda-packages/dns_validated_certificate_handler/lib/index.js

```diff
@@ -168,12 +168,38 @@ const requestCertificate = async function (requestId, domainName, subjectAlterna
  *
  * @param {string} arn The certificate ARN
  */
-const deleteCertificate = async function (arn, region) {
+const deleteCertificate = async function(arn, region) {
   const acm = new aws.ACM({ region });
 
-  console.log(`Deleting certificate ${arn}`);
-
   try {
+    console.log(`Waiting for certificate ${arn} to become unused`);
+
+    let inUseByResources;
+    const maxAttempts = 10;
+    for (let attempt = 0; attempt < maxAttempts - 1; attempt++) {
```

Comment:
I think you mean `attempt < maxAttempts`, not `attempt < maxAttempts - 1`.

Consider what happens if `maxAttempts = 1`. The following would happen:

```
for (let attempt = 0; attempt < 0; attempt++) {
}
```

And the loop body would not be executed at all.